packages/itwin/tree-widget/src/test/trees/VisibilityTreeEventHandler.test.ts

```diff
@@ -6,26 +6,36 @@
 import { expect } from "chai";
 import { EMPTY, from, Subject } from "rxjs";
 import sinon from "sinon";
-import * as moq from "typemoq";
-import { BeEvent, BeUiEvent, using } from "@itwin/core-bentley";
+import { BeEvent, using } from "@itwin/core-bentley";
 import { CheckBoxState } from "@itwin/core-react";
 import { VisibilityTreeEventHandler } from "../../components/trees/VisibilityTreeEventHandler";
 import { flushAsyncOperations } from "../TestUtils";
-import { createSimpleTreeModelNode } from "./Common";
+import { createCategoryNode, createElementNode, createModelNode, createSimpleTreeModelNode, createSubjectNode } from "./Common";
 
-import type { AbstractTreeNodeLoaderWithProvider, CheckboxStateChange, TreeModel, TreeModelChanges, TreeModelSource } from "@itwin/components-react";
-import type { IPresentationTreeDataProvider } from "@itwin/presentation-components";
+import type { AbstractTreeNodeLoaderWithProvider, CheckboxStateChange, ITreeNodeLoader } from "@itwin/components-react";
+import { MutableTreeModel , TreeModelSource } from "@itwin/components-react";
+
+import type { PresentationTreeDataProvider, PresentationTreeNodeItem } from "@itwin/presentation-components";
 import type { SelectionHandler } from "@itwin/presentation-frontend";
 import type { IVisibilityHandler, VisibilityChangeListener, VisibilityStatus, VisibilityTreeEventHandlerParams } from "../../components/trees/VisibilityTreeEventHandler";
+import { IModelApp } from "@itwin/core-frontend";
+import { waitFor } from "@testing-library/react";
 
 describe("VisibilityTreeEventHandler", () => {
-  const modelSourceMock = moq.Mock.ofType<TreeModelSource>();
-  const modelMock = moq.Mock.ofType<TreeModel>();
-  const nodeLoaderMock = moq.Mock.ofType<AbstractTreeNodeLoaderWithProvider<IPresentationTreeDataProvider>>();
-  const dataProviderMock = moq.Mock.ofType<IPresentationTreeDataProvider>();
-  const selectionHandlerMock = moq.Mock.ofType<SelectionHandler>();
+  const selectionHandlerStub = { getSelection: () => {} } as any as SelectionHandler;
+
+  const nodeLoaderStub = {
```

Comment:
I think this should created in `setupTreeModel`.

Reply:
Why do you think so?

Reply:
Because every test uses it through `setupTreeModel` and I think code would be cleaner and more readable if stub for node loader is created in `setupTreeModel` instead of having object defined at the root level, shared between all tests and being mutated before each test.